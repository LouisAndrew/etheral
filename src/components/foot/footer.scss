footer {
    margin-bottom: 0 !important;
}

.copyright {
    position: absolute;
    bottom: 16px;
    z-index: 2;
}
