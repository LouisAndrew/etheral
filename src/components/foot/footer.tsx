import React from 'react';

import { Flex, Text } from 'rebass';

import Links from './links';
import MailingList from './mailing-list';
import Social from './social';

// outer wrapper has to be styled externally as if css props is passed it would overwrite the initial variant styling
import './footer.scss';

type Props = {};

const Footer: React.FC<Props> = () => {
    return (
        <Flex variant="outerWrapper" as="footer" bg="#fff" my={[0]}>
            <Flex
                variant="innerWrapper"
                py={[5]}
                pb={[8]}
                px={[8, 8, 7]}
                flexDirection="column"
                alignItems="center"
            >
                <MailingList />
                <Links />
                <Social />
                <Text
                    className="copyright"
                    fontFamily="body"
                    width="100vw"
                    textAlign="center"
                    fontSize={[0, 0, 1]}
                    role="contentinfo"
                >
                    Copyright
                </Text>
            </Flex>
        </Flex>
    );
};

export { Footer };
