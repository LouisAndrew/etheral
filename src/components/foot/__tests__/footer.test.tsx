import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Footer } from '../footer';

describe('Footer element', () => {
    const element = <Footer />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(element, div);
    });

    it('matches snapshot', () => {
        const run = false;
        if (run) {
            const tree = renderer.create(element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
