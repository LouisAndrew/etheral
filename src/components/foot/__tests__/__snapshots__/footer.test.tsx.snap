// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer element matches snapshot 1`] = `
<footer
  className="css-jxozst"
>
  <div
    className="css-337cvv"
  >
    <div
      className="css-hhh236"
    >
      <form
        className="css-vurnku"
        onSubmit={[Function]}
      >
        <div
          className="css-171zudh"
        >
          <input
            aria-labelledby="mailing-list-input"
            className="css-4y926w"
            name="mailing-list-email"
            onChange={[Function]}
            placeholder="Insert your email here"
            type="email"
          />
          <input
            className="css-4y926w"
            type="submit"
          />
        </div>
      </form>
      <div
        className="css-unq820"
        id="mailing-list-input"
      >
        Join mailing list to be the first to receive news bout our offers
      </div>
    </div>
    <nav
      className="css-16d31i"
    >
      <a
        href="/contact"
        onClick={[Function]}
      >
        <div
          className="css-r8xm40"
        >
          CONTACT US
        </div>
      </a>
      <a
        href="/faq"
        onClick={[Function]}
      >
        <div
          className="css-r8xm40"
        >
          FAQ
        </div>
      </a>
      <a
        href="/size-guide"
        onClick={[Function]}
      >
        <div
          className="css-r8xm40"
        >
          SIZE GUIDE
        </div>
      </a>
    </nav>
    <div
      className="css-29mlpl"
    >
      <a
        href="https://www.instagram.com/etheralmade/"
        target="_blank"
      >
        <svg
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M352 0H160C71.648 0 0 71.648 0 160v192c0 88.352 71.648 160 160 160h192c88.352 0 160-71.648 160-160V160C512 71.648 440.352 0 352 0zm112 352c0 61.76-50.24 112-112 112H160c-61.76 0-112-50.24-112-112V160C48 98.24 98.24 48 160 48h192c61.76 0 112 50.24 112 112v192z"
          />
          <path
            d="M256 128c-70.688 0-128 57.312-128 128s57.312 128 128 128 128-57.312 128-128-57.312-128-128-128zm0 208c-44.096 0-80-35.904-80-80 0-44.128 35.904-80 80-80s80 35.872 80 80c0 44.096-35.904 80-80 80z"
          />
          <circle
            cx={393.6}
            cy={118.4}
            r={17.056}
          />
        </svg>
      </a>
      <a
        href="https://www.facebook.com/etheralmade"
        target="_blank"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M15.997 3.985h2.191V.169C17.81.117 16.51 0 14.996 0c-3.159 0-5.323 1.987-5.323 5.639V9H6.187v4.266h3.486V24h4.274V13.267h3.345l.531-4.266h-3.877V6.062c.001-1.233.333-2.077 2.051-2.077z"
          />
        </svg>
      </a>
      <a
        href="https://pin.it/NrIwTvY"
        target="_blank"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M12.326 0C5.747.001 2.25 4.216 2.25 8.812c0 2.131 1.191 4.79 3.098 5.633.544.245.472-.054.94-1.844a.425.425 0 00-.102-.417c-2.726-3.153-.532-9.635 5.751-9.635 9.093 0 7.394 12.582 1.582 12.582-1.498 0-2.614-1.176-2.261-2.631.428-1.733 1.266-3.596 1.266-4.845 0-3.148-4.69-2.681-4.69 1.49 0 1.289.456 2.159.456 2.159S6.781 17.4 6.501 18.539c-.474 1.928.064 5.049.111 5.318.029.148.195.195.288.073.149-.195 1.973-2.797 2.484-4.678.186-.685.949-3.465.949-3.465.503.908 1.953 1.668 3.498 1.668 4.596 0 7.918-4.04 7.918-9.053C21.733 3.596 17.62 0 12.326 0z"
          />
        </svg>
      </a>
    </div>
    <div
      className="copyright css-3u5913"
      role="contentinfo"
    >
      Copyright
    </div>
  </div>
</footer>
`;
